godtossing/Pos/Ssc/GodTossing/Arbitrary.hs

```diff
@@ -103,18 +107,22 @@ commitmentsAndOpenings =
     vssPk = toVssPublicKey $ deterministicVssKeyGen "ababahalamaha"
 {-# NOINLINE commitmentsAndOpenings #-}
 
-
 instance Arbitrary CommitmentOpening where
     arbitrary = elements commitmentsAndOpenings
+    shrink = genericShrink
 
 instance Nonrepeating CommitmentOpening where
     nonrepeating n = sublistN n commitmentsAndOpenings
 
 instance Arbitrary Commitment where
     arbitrary = coCommitment <$> arbitrary
+    shrink Commitment {..} = [ Commitment { commShares = shrunkShares, .. }
```

Comment:
How do you know that this shrinking preserves the invariant of the data type (see comment where `Commitment` is defined)?

```
'checkSig vcSigningKey (vcVssKey, vcExpiryEpoch) vcSignature'.
```